### PR TITLE
[18.0][MIG] delivery_package_fee: Migration to 18.0

### PR DESCRIPTION
**Pull request by chaule97**
- Since the `delivery` module in [version 15.0](https://github.com/odoo/odoo/pull/110686/files) depends on `sale` instead of `sale_stock`, I need to update the `delivery_package_fee` module to depend on `stock_delivery` instead of `delivery`.

### CODE REVIEW

File: delivery_package_fee/__manifest__.py [hunk 1]
```diff
@@ -3,12 +3,12 @@
...
     "name": "Delivery Package Fees",
     "summary": "Add fees on delivered packages on shipping methods",
-    "version": "16.0.1.1.1",
+    "version": "18.0.1.0.0",
     "category": "Delivery",
     "website": "https://github.com/OCA/delivery-carrier",
     "author": "Camptocamp, Odoo Community Association (OCA)",
     "installable": True,
     "license": "AGPL-3",
-    "depends": ["delivery"],
```
Comment (trisdoan) [inline delivery_package_fee/__manifest__.py]: I am still confused why you did this

File: delivery_package_fee/tests/test_package_fee.py [hunk 2]
```diff
@@ -10,10 +11,15 @@ class TestPackageFee(TransactionCase):
     def setUpClass(cls):
         super().setUpClass()
         cls.product1 = cls.env["product.product"].create(
-            {"name": "Product 1", "type": "product", "lst_price": 1.0}
+            {"name": "Product 1", "is_storable": True, "lst_price": 1.0}
```
Comment (trisdoan) [inline delivery_package_fee/tests/test_package_fee.py:14]: why?

File: delivery_package_fee/tests/test_package_fee.py [hunk 3]
```diff
@@ -10,10 +11,15 @@ class TestPackageFee(TransactionCase):
...
         )
         cls.product2 = cls.env["product.product"].create(
-            {"name": "Product 2", "type": "product", "lst_price": 1.0}
+            {"name": "Product 2", "is_storable": True, "lst_price": 1.0}
+        )
+        cls.pricelist = cls.env["product.pricelist"].create(
+            {
+                "name": "Test Pricelist",
+            }
```
Comment (trisdoan) [inline delivery_package_fee/tests/test_package_fee.py:22]: why?

File: delivery_package_fee/tests/test_package_fee.py [hunk 4]
```diff
@@ -94,10 +100,12 @@ def test_package_fee_simple(self):
         """All stock moves processed at once"""
         picking = self.sale.picking_ids
         self.assertEqual(picking.state, "assigned")
-        picking.move_line_ids[0].result_package_id = self.pack1
-        picking.move_line_ids[0].qty_done = 10.0
-        picking.move_line_ids[1].result_package_id = self.pack2
-        picking.move_line_ids[1].qty_done = 10.0
+        picking.move_line_ids[0].write(
+            {"result_package_id": self.pack1, "quantity": 10.0, "picked": True}
+        )
```
Comment (trisdoan) [inline delivery_package_fee/tests/test_package_fee.py:105]: why using quantity and picked to replace qty_done?

File: delivery_package_fee/__manifest__.py (hunk 1 above)
Comment (chaule97) [reply delivery_package_fee/__manifest__.py]: In this module, we need to inherit `stock.picking`, but the current module `delivery` does not depend on `sale_stock`. Therefore, I have updated `delivery_package_fee` to depend on `stock_delivery`.

File: delivery_package_fee/tests/test_package_fee.py (hunk 2 above)
Comment (chaule97) [reply delivery_package_fee/tests/test_package_fee.py:14]: the stock module version 17.0, the `type` field in the `product` model includes an additional value, [product](https://github.com/odoo/odoo/blob/f118a44c5b15163556e3388c5a7daa72119f58f8/addons/stock/models/product.py#L661). However, in version 18.0, this has been replaced by the [is_storable](https://github.com/odoo/odoo/blob/a152910e3a352516ea1240d8833a406b144382dd/addons/stock/models/product.py#L677) field

File: delivery_package_fee/tests/test_package_fee.py (hunk 3 above)
Comment (chaule97) [reply delivery_package_fee/tests/test_package_fee.py:22]: in `test_package_fee_pricelist`, we retrieve the [pricelist](https://github.com/chaule97/delivery-carrier/blob/1b2e553e2ef943dbb5d848a1a0d90f5d0501c39d/delivery_package_fee/tests/test_package_fee.py#L329) from `sale.order` . The pricelist in `sale.order` is from [order.partner_id.property_product_pricelist](https://github.com/odoo/odoo/blob/a152910e3a352516ea1240d8833a406b144382dd/addons/sale/models/sale_order.py#L438). which is from [_get_partner_pricelist_multi](https://github.com/odoo/odoo/blob/a152910e3a352516ea1240d8833a406b144382dd/addons/product/models/res_partner.py#L34C45-L34C73).

in [_get_partner_pricelist_multi](https://github.com/odoo/odoo/blob/a152910e3a352516ea1240d8833a406b144382dd/addons/product/models/product_pricelist.py#L284), my understanding is that it selects the first active pricelist  (`active=True`) with the same `company_id`. Currently, in the test environment, there are no records in the pricelist table, so I need to create one.

File: delivery_package_fee/tests/test_package_fee.py (hunk 4 above)
Comment (chaule97) [reply delivery_package_fee/tests/test_package_fee.py:105]: https://github.com/odoo/odoo/pull/137864, we don't have `qty_done` anymore, we will use `quantity` and `picked`.

File: delivery_package_fee/__manifest__.py (hunk 1 above)
Comment (trisdoan) [reply delivery_package_fee/__manifest__.py]: ok, so you can update the description something like:

> From Odoo 17.0+, stock was separated from delivery, in https://github.com/odoo/odoo/pull/110686/commits/b2541e6218e014a3ed1f2b58915ad897eb2a3f58, hence delivery_package_fee depends on stock_delivery

File: delivery_package_fee/tests/test_package_fee.py (hunk 2 above)
Comment (trisdoan) [reply delivery_package_fee/tests/test_package_fee.py:14]: ok, would you mind updating the description as well, smt like

From Odoo 18.0, is_storable was introduced for simpler tracking, in https://github.com/odoo/odoo/pull/165372/commits/83d0f2bb33765f8840821c11a269c48ca28a5731

File: delivery_package_fee/tests/test_package_fee.py (hunk 3 above)
Comment (trisdoan) [reply delivery_package_fee/tests/test_package_fee.py:22]: ok, can you update the description as well

From Odoo 17.0, Public pricelist was removed in https://github.com/odoo/odoo/pull/84920/commits/40f0482b680a7872d338fa8fd10eec4663637b46, adding pricelist for testing

File: delivery_package_fee/tests/test_package_fee.py (hunk 4 above)
Comment (trisdoan) [reply delivery_package_fee/tests/test_package_fee.py:105]: ok, could you please mention it in PR's description

File: delivery_package_fee/tests/test_package_fee.py [hunk 5]
```diff
@@ -543,17 +563,22 @@ def test_package_with_type_one_fee_line(self):
         )
 
     def test_package_with_type_one_fee_line_qty(self):
-        """Assign all fee lines to a package type, but one fee line added to SO, qty=2"""
+        """
+        Assign all fee lines to a package type
+        but one fee line added to SO, qty=2
+        """
```
Comment (trisdoan) [inline delivery_package_fee/tests/test_package_fee.py:569]: pre-commit?

**Review by trisdoan — APPROVED** [review]

File: delivery_package_fee/tests/test_package_fee.py (hunk 5 above)
Comment (chaule97) [reply delivery_package_fee/tests/test_package_fee.py:569]: yes, it set that

**Pull request merged**